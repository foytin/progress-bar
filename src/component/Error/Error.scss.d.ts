declare const styles: {
  readonly centered: string;
  readonly emoji: string;
  readonly title: string;
  readonly text: string;
};

export = styles;
