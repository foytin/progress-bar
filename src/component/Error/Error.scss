$font-color: #555;

.centered {
  text-align: center;
}

.emoji {
  font-size: 5em;
  user-select: none;
}

.title {
  color: $font-color;
  font-size: 2.5rem;
  text-align: center;
}

.text {
  color: $font-color;
  font-size: 2.5rem;
  text-align: center;
}
