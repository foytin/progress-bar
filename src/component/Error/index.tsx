import React from "react";
import styles from "./Error.scss";

const Error = (): React.ReactElement => (
  <div className={styles.centered} data-testid="error-page">
    <span role="img" className={styles.emoji} aria-label="error">
      ðŸ˜­
    </span>
    <p className={styles.title}>Ooooops!</p>
    <p className={styles.text}>Out of Service! Please try it later.</p>
  </div>
);

export default Error;
