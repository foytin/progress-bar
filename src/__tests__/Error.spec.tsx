import React from "react";
import { render } from "@testing-library/react";

import Error from "../component/Error";

describe("Error Page component", () => {
  it("Error page render correctly", () => {
    const wrapper = render(<Error />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
