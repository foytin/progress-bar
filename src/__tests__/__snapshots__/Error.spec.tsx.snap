// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page component Error page render correctly 1`] = `
<div>
  <div
    class="centered"
    data-testid="error-page"
  >
    <span
      aria-label="error"
      class="emoji"
      role="img"
    >
      ðŸ˜­
    </span>
    <p
      class="title"
    >
      Ooooops!
    </p>
    <p
      class="text"
    >
      Out of Service! Please try it later.
    </p>
  </div>
</div>
`;
